x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/add_prebuilt_rules_table/add_prebuilt_rules_table.test.tsx

```diff
@@ -0,0 +1,299 @@
+  useUiSetting$: jest.fn().mockReturnValue([false]),
+  useKibana: jest.fn().mockReturnValue({
+    services: {
+      docLinks: { links: { siem: { ruleChangeLog: '' } } },
+    },
+  }),
+}));
+
+jest.mock('../../../../../common/components/links', () => ({
+  useGetSecuritySolutionLinkProps: () =>
+    jest.fn().mockReturnValue({
+      onClick: jest.fn(),
+    }),
+}));
+
+jest.mock(
+  '../../../../rule_management/api/hooks/prebuilt_rules/use_fetch_prebuilt_rules_status_query',
+  () => ({
+    useFetchPrebuiltRulesStatusQuery: jest.fn().mockReturnValue({
+      data: {
+        prebuiltRulesStatus: {
+          num_prebuilt_rules_total_in_package: 1,
+        },
+      },
+    }),
+  })
+);
+
+jest.mock('../../../../rule_management/logic/use_upgrade_security_packages', () => ({
+  useIsUpgradingSecurityPackages: jest.fn().mockImplementation(() => false),
+}));
+
+jest.mock(
+  '../../../../rule_management/logic/prebuilt_rules/use_prebuilt_rules_install_review',
+  () => ({
+    usePrebuiltRulesInstallReview: jest.fn().mockReturnValue({
+      data: {
+        rules: [
+          {
+            id: 'rule-1',
+            name: 'rule-1',
+            tags: [],
+            risk_score: 1,
+            severity: 'low',
+          },
+        ],
+        stats: {
+          num_rules_to_install: 1,
+          tags: [],
+        },
+      },
+      isLoading: false,
+      isFetched: true,
+    }),
+  })
+);
+
+jest.mock('../../../../../detections/components/user_info', () => ({
+  useUserData: jest.fn(),
+}));
+
+describe('AddPrebuiltRulesTable', () => {
+  it('disables `Install all` button if user has no write permissions', async () => {
+    (useUserData as jest.Mock).mockReturnValue([
+      {
+        loading: false,
+        canUserCRUD: false,
+      },
+    ]);
+
+    render(
+      <AddPrebuiltRulesTableContextProvider>
+        <AddPrebuiltRulesHeaderButtons />
+        <AddPrebuiltRulesTable />
+      </AddPrebuiltRulesTableContextProvider>
+    );
+
+    const installAllButton = screen.getByTestId('installAllRulesButton');
+
+    expect(installAllButton).toHaveTextContent('Install all');
+    expect(installAllButton).toBeDisabled();
+  });
+
+  it('disables `Install all` button if prebuilt package is being installed', async () => {
+    (useUserData as jest.Mock).mockReturnValue([
+      {
+        loading: false,
+        canUserCRUD: true,
+      },
+    ]);
+
+    (useIsUpgradingSecurityPackages as jest.Mock).mockReturnValueOnce(true);
+
+    render(
+      <AddPrebuiltRulesTableContextProvider>
+        <AddPrebuiltRulesHeaderButtons />
+        <AddPrebuiltRulesTable />
+      </AddPrebuiltRulesTableContextProvider>
+    );
+
+    const installAllButton = screen.getByTestId('installAllRulesButton');
+
+    expect(installAllButton).toHaveTextContent('Install all');
+    expect(installAllButton).toBeDisabled();
+  });
+
+  it('renders Install all` button when user has permissions', async () => {
```

Comment:
Should it be
"Enables `Install all` button when user has write permissions"
?

Reply:
Also, do you think it would be a good idea to test a case that's causing the bug: `canUserCRUD` is `false` -> user sees `There are no prebuilt detection rules available for installation`?

Reply:
This scenario should look the same for users independently of their Security permissions. I'll create two separate tests for the use case of no rules with permissions and no rules without permissions. Results should be the same.